### ISSUE TITLE
Multiple schemas in same database pgsql

### ISSUE THREAD

**Issue by qpautrat** [cakephp/phinx#709]
Hello,

I'm wondering if Phinx are able to handle multiple schemas for given database with PostgreSQL ?

I want to do something like this:

``` php
$this->createTable('foo.bar')
```

where `foo` is one of my schema and `bar` the table name.

Thanks !

**Comment by BallisticPain** [comment]
@qpautrat 
Have you tried this? I just peeked down in the code, and I don't really see any reason why it wouldn't work.

**Comment by qpautrat** [comment]
Hi @BallisticPain ,

To be fair, it's a long time ago, i don't remember exactly, but i guess i tried and it didn't work.

**Comment by BallisticPain** [comment]
I assumed you had, but thought I would ask. I didn't see anything that would lead me to believe it's not possible, but it may have more to do with configuration. If it's not already a feature, then you could always bring a PR for it. ;)

Sorry I wasn't of any assistance.

**Comment by renegat59** [comment]
I have figured out that you can define the schema parameter in the phinx configuration, however, there is an issue there. I made a new schema called "blog" and set it in phinx config. The first migration (creating all the tables) went ok, and phinx created it's own table with the migration history or sth, but once I wanted to run another migration (alter table) it was complaining that the phinx table was not there, since he was searching for it in schema 'public'. I will recreate the issue tonight and paste here the exact errors and steps to recreate.

**Comment by paulop** [comment]
Hello,
I have this code in class extended of AbstractMigration

```
public function change()
    {        
        //select schemaname
        $schemas = $this->fetchAll("SELECT id, schemaname FROM empresas WHERE TRIM(schemaname) <> '' ;");        
        
        foreach($schemas as $schema){
            $schemaname = $schema['schemaname'];
            
            $this->table("recorrencias", ['schema'=>$schemaname])
            ->addColumn('ativo', 'boolean', [
                'default' => true,
                'null' => false,
            ])
            ->addColumn('convenio', 'string', [
                'default' => null,
                'limit' => 20,
                'null' => true,
            ])
            ->update();
```
doesn't work

and this

```
$this->table("$schemaname.recorrencias")
            ->addColumn('ativo', 'boolean', [
                'default' => true,
                'null' => false,
            ])
            ->addColumn('convenio', 'string', [
                'default' => null,
                'limit' => 20,
                'null' => true,
            ])
            ->update();
```

doesn't work too

```
$this->table("$schemaname.recorrencias")->exists();
//return false
```

Any help for update multiple schemas?

**Comment by paulop** [comment]
I solved using this
`$this->getAdapter()->setOptions(array_replace($this->getAdapter()->getOptions(), ['schema' => $schemaname]));`

**Comment by leovant** [comment]
I had to use @paulop workaround too. Using 0.8.1.

**Comment by dereuromark** [comment]
Are you able to provide a patch with your suggested changes as PR?
Otherwise please open a PR for documentation improvement.

**Comment by paulop** [comment]
@dereuromark In the documentation, you say to write an example here?

https://book.cakephp.org/3.0/en/phinx/migrations.html#creating-a-table

**Comment by dereuromark** [comment]
Yes :)

**Comment by corpsee** [comment]
This feature implemented in #1193.

**Comment by othercorey** [comment]
Added in https://github.com/cakephp/phinx/pull/1193
